plugins/inputs/tencentcloudcm/tencentcloudcm.go

```diff
@@ -0,0 +1,448 @@
+		# metrics = ["CPUUsage", "MemUsage"]
+  
+		[[inputs.tencentcloudcm.accounts.namespaces.regions]]
+		  # Tencent Cloud regions (required - Allowed values: https://intl.cloud.tencent.com/document/api/248/33876)
+		  region = "ap-guangzhou"
+  
+		  ## Dimension filters for Metric. Different namespaces may have different
+		  ## dimension requirements, e.g. CVM Monitoring Metrics: https://intl.cloud.tencent.com/document/product/248/6843It must be specified if the namespace does not support instance auto discovery
+		  ## Currently, discovery supported for the following namespaces:
+		  ## - QCE/CVM
+		  ## - QCE/CDB
+		  ## - QCE/CES
+		  ## - QCE/REDIS
+		  ## - QCE/LB_PUBLIC
+		  ## - QCE/LB_PRIVATE
+		  # [[inputs.tencentcloudcm.accounts.namespaces.regions.instances]]
+		  # [[inputs.tencentcloudcm.accounts.namespaces.regions.instances.dimensions]]
+		  #   name = ""
+		  #   value = ""
+`
+}
+
+// Description implements telegraf.Input interface
+func (t *TencentCloudCM) Description() string {
+	return "Pull Metric Statistics from Tencent Cloud Cloud Monitor"
+}
+
+// Init is for setup, and validating config.
+func (t *TencentCloudCM) Init() error {
+
+	if t.Period <= 0 {
+		return fmt.Errorf("period is empty")
+	}
+
+	if len(t.Accounts) == 0 {
+		return errors.New("account is empty")
+	}
+
+	// create account credential
+	for i := range t.Accounts {
+
+		if t.Accounts[i].SecretID == "" {
+			return fmt.Errorf("secret_id is empty")
+		}
+		if t.Accounts[i].SecretKey == "" {
+			return fmt.Errorf("secret_key is empty")
+		}
+
+		t.Accounts[i].Crs = common.NewCredential(
+			t.Accounts[i].SecretID,
+			t.Accounts[i].SecretKey,
+		)
+		if t.Accounts[i].Name == "" {
+			t.Accounts[i].Name = fmt.Sprintf("%v", i)
+		}
+		// check if namespace supports auto discovery
+		for _, namespace := range t.Accounts[i].Namespaces {
+			if namespace.Name == "" {
+				return fmt.Errorf("namespace is empty")
+			}
+			for _, region := range namespace.Regions {
+				if region.RegionName == "" {
+					return fmt.Errorf("region is empty")
+				}
+				_, ok := Registry[namespace.Name]
+				if len(region.Instances) == 0 && !ok {
+					return fmt.Errorf(
+						"unsupported namespace %s for discovering instances, please specify instances and dimensions",
+						namespace.Name,
+					)
+				}
+			}
+		}
+	}
+
+	// start discovery
+	t.discoverTool = NewDiscoverTool(t.Log)
+	go t.discoverTool.Discover(t.Accounts, t.DiscoveryInterval, t.Endpoint)
+	t.discoverTool.DiscoverMetrics()
+	for i := range t.Accounts {
+		for j := range t.Accounts[i].Namespaces {
+			// use discovered metrics if not specified
+			if len(t.Accounts[i].Namespaces[j].Metrics) == 0 {
+				metrics, ok := t.discoverTool.DiscoveredMetrics[t.Accounts[i].Namespaces[j].Name]
+				if !ok {
+					errorInfo := fmt.Sprintf("unsupported namespace %s for discovering metrics, please specify metrics",
+						t.Accounts[i].Namespaces[j].Name)
+					t.Log.Error(errorInfo)
+					return errors.New(errorInfo)
```

Comment:
The code as is, will likely add the error message twice to the log file as the returned error will also be logged. Furthermore, I guess we do not want to accept "faulty" accounts, so just returning the error is perfectly fine, no need for logging it separately.
You can then simplify to
```suggestion
					return fmt.Errorf("unsupported namespace %s for discovering metrics, please specify metrics",
						t.Accounts[i].Namespaces[j].Name)
```